docs/guides/integration-javascript-client-side.md

```diff
@@ -326,32 +338,39 @@ window.__uid2.callbacks.push(async (eventType, payload) => {
         // }
         var advertising_token_to_use = payload.identity.advertising_token;
       } else {
-          if (__uid2.isLoginRequired()) {
+          if (__uid2.hasOptedOut()) {
+            // DII has opted out of UID2, no UID2 token will be generated
+          }
+          else if (__uid2.isLoginRequired()) {
             // Call one of the setIdentityFrom functions to generate a new UID2 token.
             // Add any retry logic around this call as required.
             await __uid2.setIdentityFromEmailHash(
                 emailHash,
-                clientSideConfig
-          );
+                clientSideConfig);
+          }  
           else {
             // there is a token generation API call in flight which triggers
             // a IdentityUpdated event 
           }
-        }
       }
       break;
  
     case "IdentityUpdated":
       // The IdentityUpdated event happens when a UID2 token is generated or refreshed.
       // See previous comment for an example of how the payload looks.
-      var advertising_token_to_use = payload.identity.advertising_token;
+      // It's possible that payload.identity is null because the DII has opted out of UID2
+      if (!payload.identity) {
```

Comment:
if the user has opted out, the payload should not have a identity and we should add this condition to avoid misleading pubs to blindly get ```payload.identity.advertising_token```

If you agree, please check syntax tho

Reply:
It could be null for other reasons as well - the user logged out, the token expired without being refreshed, maybe other things. And I'm not convinced the identity should be null if the user has logged out.

If they want an advertising token, they should check for it:
if (payload?.identity?.advertising_token) {

Reply:
ok adjusted as you suggested.